backends/jwt_remote.go

```diff
@@ -323,3 +360,41 @@ func (o *remoteJWTChecker) jwtRequest(host, uri, token string, dataMap map[strin
 	log.Debugf("jwt request approved for %s", token)
 	return true, nil
 }
+
+func (o *remoteJWTChecker) getHost(token string) (string, error) {
+	if o.host != "" {
+		return o.host, nil
+	}
+
+	// Actually this should never happen because of configuration sanity check. TODO: consider removing this condition.
+	if !o.options.parseToken {
+		errorString := fmt.Sprintf("impossible to obtain host for the authorization request - token parsing is turned off")
+		return "", errors.New(errorString)
+	}
+
+	iss, err := getIssForToken(o.options, token, o.options.skipUserExpiration)
+	if err != nil {
+		errorString := fmt.Sprintf("cannot obtain host for the authorization request from token %s: %s", token, err)
+		return "", errors.New(errorString)
+	}
+
+	if !o.isHostWhitelisted(iss) {
+		errorString := fmt.Sprintf("host %s obtained from host is not whitelisted; rejecting", iss)
+		return "", errors.New(errorString)
+	}
+
+	return iss, nil
+}
+
+func (o *remoteJWTChecker) isHostWhitelisted(host string) bool {
+	if len(o.hostWhitelist) == 1 && o.hostWhitelist[0] == whitelistMagicForAnyHost {
+		return true
+	}
+
+	for _, whitelistedHost := range o.hostWhitelist {
```

Comment:
I just realized we could optimize this by having a `var hostsWhitelist map[string]struct{}` instead of a slice of them to get `O(1)` lookups, but I don't see the array being too big, so maybe not worth it for now.